sympy/combinatorics/free_group.py

```diff
@@ -806,6 +831,58 @@ def subword(self, from_i, to_j):
             array_form = letter_form_to_array_form(letter_form, group)
             return group.dtype(array_form)
 
+    def is_independent(self, word):
```

Comment:
That is fine. I think I already understood everything you mentioned in 
your comment, (even your comment has been executed in code 
successfully). But the problem is about the removal of `independent` 
relator from the relator list. See for example see this doctest example 
<https://github.com/sympy/sympy/pull/11295/files#diff-2bca05fcc1cad2237c5589d92187d25cR1288> 
. Did I missed something?

On 06/30/2016 06:04 PM, Kalevi Suominen wrote:
>
> In sympy/combinatorics/free_group.py 
> <https://github.com/sympy/sympy/pull/11295#discussion_r69123414>:
>
> > @@ -806,6 +831,58 @@ def subword(self, from_i, to_j):
> >              array_form = letter_form_to_array_form(letter_form, group)
> >              return group.dtype(array_form)
> >
> > +    def is_independent(self, word):
>
> It seems that the relators are scanned and generators that appear only 
> once are marked redundant. Those relators that contain such generators 
> are skipped over. After all relators have been scanned redundant 
> generators are replaced in the relators by their representations in 
> terms of other generators and deleted from the list. Then the process 
> can be repeated. (Relators containing generators marked redundant 
> cannot be used to decide if a generator is redundant. Even if it 
> appears only once, it may reappear when another generator is replaced.)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/sympy/sympy/pull/11295/files/bb2622cc696920381b63f2d02e4a5e928dc221e5#r69123414>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AHWt8f3hKesJ76PVUKgenMpwxGtQN7jSks5qQ7fmgaJpZM4I-QZs>.
>